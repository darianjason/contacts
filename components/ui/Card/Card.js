// @flow

import * as React from 'react';
import {View} from 'react-native';
import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet';

import styles from './Card.styles';

type Props = {
  style?: ViewStyleProp,
  children: ?React.Node,
};

const Card = ({style, children}: Props): React.Node => (
  <View style={[styles.card, style]}>{children}</View>
);

export default Card;
