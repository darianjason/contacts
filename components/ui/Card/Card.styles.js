import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  card: {
    borderRadius: 8,
    overflow: 'hidden',
  },
});

export default styles;
