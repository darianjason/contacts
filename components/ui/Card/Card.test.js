import React from 'react';
import {Text} from 'react-native';
import {render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('Card component', () => {
  test('renders children', () => {
    const {getByText} = render(
      <Card>
        <Text>Test</Text>
      </Card>,
    );

    expect(getByText('Test')).toBeTruthy();
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Card>
          <Text>Test</Text>
        </Card>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
