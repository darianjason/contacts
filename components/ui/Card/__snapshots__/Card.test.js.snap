// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 8,
      "overflow": "hidden",
    }
  }
>
  <Text>
    Test
  </Text>
</View>
`;
